### PR TITLE
add sim_sir_df

### PR DESCRIPTION
**Pull request by jlubken**
Add `sim_sir_df(...) -> pd.DataFrame`
Eliminate multiple implicit int casts to float in `sir(...)` comparisons.
Eliminate tuple unpacking of `y` in `sir(...)`.
Add test for `sim_sir_df(...)`.
Add type annotations.
Allow variadic number of rates to be passed to get_hospitalizations2.

Refers to #108

### CODE REVIEW

File: penn_chime/models.py [hunk 1]
```diff
@@ -1,49 +1,93 @@
...
+    return pd.DataFrame(
+        data=gen_sir(s, i, r, beta, gamma, n_days, beta_decay),
+        columns=("S", "I", "R"),
+    )
+
+
+@st.cache
+def get_hospitalizations2(
+    infected: np.ndarray, rates: Tuple[float, ...], market_share: float = 1.0
+) -> Tuple[np.ndarray, ...]:
```
Comment (quinn-dougherty) [inline penn_chime/models.py:81]: is this return sign `Tuple[np.ndarray, np.ndarray, np.ndarray]`?

**Comment by quinn-dougherty** [comment]
This PR also does https://github.com/CodeForPhilly/chime/issues/61

File: penn_chime/models.py (hunk 1 above)
Comment (jlubken) [reply penn_chime/models.py:81]: It returns as many np.ndarray as there are floats in the rates tuple. This way the code is not dependent on there being three things (hospitalized, icu, ventilated).

**Review by quinn-dougherty — APPROVED** [review]
code looks 100% and i did side-by-side verification against prod locally.

**Pull request merged**